### PR TITLE
Remove unused host functions (Fixes #231)

### LINKED ISSUE #231: Remove `host_bindings_loose.rs` and `host_bindings_loose_for_testing.rs`
These are no longer needed (I think).

### PR DESCRIPTION
**Pull request by sappenin**
## High Level Overview of Change

This PR removes the no-longer-used/needed ` host_bindings_loose.rs` and `host_bindings_loose_for_testing.rs`. During implementation of this issue, it was discovered that the `projects/e2e_tests/codecov_tests` were not being executed as part of the `scripts/e2e-tests.sh` build step (because the `codecov_tests` lacked a `fixtures` folder, prompting that test to be ignored).  This PR also restores that test and updates assertions based on real fixture data.

### Type of Change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Refactor (non-breaking change that only restructures code)
- [X] Tests (You added tests for code that already exists, or your new feature included in this PR)

### CODE REVIEW

File: projects/e2e-tests/codecov_tests/src/lib.rs [hunk 1]
```diff
@@ -281,39 +281,35 @@ pub extern "C" fn finish() -> i32 {
...
-        19,
+        62,
         "trace_amount",
     );
 
     // ########################################
     // Step #2: Test set_data edge cases
     // ########################################
-    check_result(
-        unsafe { host_bindings_loose::get_parent_ledger_hash(-1, 4) },
```
Comment (sappenin) [inline projects/e2e-tests/codecov_tests/src/lib.rs:292]: Passing a negative number to `get_parent_ledger_hash` is not impossible per Rust type-checking.

File: projects/e2e-tests/codecov_tests/src/lib.rs [hunk 2]
```diff
@@ -281,39 +281,35 @@ pub extern "C" fn finish() -> i32 {
...
         )
     });
-    with_buffer::<4, _, _>(|ptr, _len| {
-        check_result(
-            unsafe { host_bindings_loose::get_parent_ledger_hash(ptr as i32, 1_000_000_000) },
-            error_codes::POINTER_OUT_OF_BOUNDS,
-            "get_parent_ledger_hash_len_too_long",
-        )
-    });
+    // NOTE: This test is commented out because the WASM runtime catches the out of bounds
```
Comment (sappenin) [inline projects/e2e-tests/codecov_tests/src/lib.rs:298]: LMK and I'll simply remove these commented out tests. Wanted to leave them in for now to make sure we evaluate this change-set properly.

File: projects/e2e-tests/codecov_tests/src/lib.rs [hunk 3]
```diff
@@ -353,114 +350,97 @@ pub extern "C" fn finish() -> i32 {
...
-        unsafe {
-            host_bindings_loose::get_tx_nested_array_len(
-                locator.as_ptr() as i32 + 1_000_000_000,
-                locator.len() as i32,
-            )
-        },
-        error_codes::POINTER_OUT_OF_BOUNDS,
-        "get_tx_nested_array_len_ptr_oob",
-    );
+    // NOTE: These tests are commented out because the WASM runtime catches the out of bounds
```
Comment (sappenin) [inline projects/e2e-tests/codecov_tests/src/lib.rs:359]: LMK and I'll simply remove these commented out tests. Wanted to leave them in for now to make sure we evaluate this change-set properly.

File: wasm-host-simulator/src/host_functions_wamr.rs [hunk 4]
```diff
@@ -887,6 +887,162 @@ pub fn get_nft(
...
+
+pub fn get_ledger_account_hash(
+    env: wasm_exec_env_t,
+    out_buf_ptr: *mut u8,
+    out_buf_cap: usize,
+) -> i32 {
+    if HASH256_LEN > out_buf_cap {
+        return HostError::BufferTooSmall as i32;
+    }
+    // For now, return a zero hash as a stub implementation
```
Comment (sappenin) [inline wasm-host-simulator/src/host_functions_wamr.rs:939]: Unclear if these stub implementations are sufficient, or if we want to add something to the `wasm-host-simulator` here? Currently, I lean towards "do nothing except use these stubs" but could be persuaded otherwise.

File: projects/e2e-tests/codecov_tests/src/lib.rs [hunk 5]
```diff
@@ -17,15 +17,6 @@ use xrpl_wasm_std::host::error_codes;
 use xrpl_wasm_std::host::trace::{trace, trace_num as trace_number};
```
[the remark this answers is not in the document]
Comment (sappenin) [reply projects/e2e-tests/codecov_tests/src/lib.rs:17]: This is now restored.
